File: app/src/main/java/com/example/findmyrahmah/restaurantAdapter.java
package com.example.findmyrahmah;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

public class restaurantAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {

    private static final int TYPE = 1;
    private Context context;
    private List<Object> listRecyclerItem;


    public void RecyclerAdapter(Context context, List<Object> listRecyclerItem){
        this.context = context;
        this.listRecyclerItem = listRecyclerItem;
    }

    public class ItemViewHolder extends RecyclerView.ViewHolder {
        private TextView restaurantName;
        private TextView restaurantAddress;
        private TextView restaurantRating;

        public ItemViewHolder (@NonNull View itemView){
            super(itemView);
            restaurantName = (TextView) itemView.findViewById(R.id.RestaurantName);
            restaurantAddress = (TextView) itemView.findViewById(R.id.RestaurantAddress);
            restaurantRating = (TextView) itemView.findViewById(R.id.RestaurantRating);
        }
    }

    @NonNull
    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        switch (i){
            case TYPE:
            default:
                View layoutView = LayoutInflater.from(viewGroup.getContext()).inflate(R.layout.restaurant, viewGroup, false);

                return new ItemViewHolder((layoutView));
        }
    }

    @Override
    public void onBindViewHolder(@NonNull RecyclerView.ViewHolder viewHolder, int i) {
        int viewType = getItemViewType(i);

        switch (viewType){
            case TYPE:
            default:
                ItemViewHolder itemViewHolder = (ItemViewHolder) viewHolder;

                Restaurant restaurant = (Restaurant) listRecyclerItem.get(i);

                itemViewHolder.restaurantName.setText(restaurant.getRestaurantName());
                itemViewHolder.restaurantAddress.setText(restaurant.getRestaurantAddress());
                itemViewHolder.restaurantRating.setText(restaurant.getRestaurantRating().toString());
        }
    }

    @Override
    public int getItemCount() {
        return listRecyclerItem.size();
    }
}

File: app/src/main/java/com/example/findmyrahmah/RestaurantAdapter.kt
package com.example.findmyrahmah

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.ImageView
import android.widget.TextView
import androidx.lifecycle.MutableLiveData
import androidx.recyclerview.widget.RecyclerView

class RestaurantAdapter(private val restaurantList: MutableLiveData<List<Restaurant>>) : RecyclerView.Adapter<RestaurantAdapter.RestaurantViewHolder>() {

    inner class RestaurantViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        private val imageView: ImageView = itemView.findViewById(R.id.RestaurantView)
        private val nameTextView: TextView = itemView.findViewById(R.id.RestaurantName)
        private val addressTextView: TextView = itemView.findViewById(R.id.RestaurantAddress)
        private val ratingTextView: TextView = itemView.findViewById(R.id.RestaurantRating)
        private val likeTextView: TextView = itemView.findViewById(R.id.TotalLike)
        private val button: Button = itemView.findViewById(R.id.likeButton)

        fun bind(restaurant: Restaurant) {
            imageView.setImageResource(restaurant.image)
            nameTextView.text = restaurant.name
            addressTextView.text = restaurant.address
            ratingTextView.text = restaurant.rating.toString()
            likeTextView.text = restaurant.likeCount.toString()
            button.isEnabled = restaurant.isOnline
            button.setOnClickListener {
                //increase like count
                restaurant.likeCount++

                //Update UI with new like count
                likeTextView.text = "${restaurant.likeCount}"
            }
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RestaurantViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.restaurant, parent, false)
        return RestaurantViewHolder(view)
    }

    override fun onBindViewHolder(holder: RestaurantViewHolder, position: Int) {
        holder.bind(restaurantList[position])
    }

    override fun getItemCount(): Int {
        return restaurantList.size
    }
}


